templates/directory.php

```diff
@@ -112,7 +112,7 @@ function pmpromd_shortcode($atts, $content=null, $code="")
 $sql_parts['LIMIT'] = "LIMIT $start, $limit";
 
 if( $s ) {
-	$sql_parts['WHERE'] .= "AND (u.user_login LIKE '%" . esc_sql($s) . "%' OR u.user_email LIKE '%" . esc_sql($s) . "%' OR u.display_name LIKE '%" . esc_sql($s) . "%' OR um.meta_value LIKE '%" . esc_sql($s) . "%') ";
+	$sql_parts['WHERE'] .= apply_filters( "pmpro_member_directory_search_sql", "AND (u.user_login LIKE '%" . esc_sql($s) . "%' OR u.user_email LIKE '%" . esc_sql($s) . "%' OR u.display_name LIKE '%" . esc_sql($s) . "%' OR um.meta_value LIKE '%" . esc_sql($s) . "%') ", $s );
```

Comment:
Note -- I'm using a better name for the hookdoc parameter because it reads better for those implementing their own hooks.

```suggestion
	$sql_where_search = "
		AND (
			u.user_login LIKE '%" . esc_sql( $s ) . "%'
			OR u.user_email LIKE '%" . esc_sql( $s ) . "%'
			OR u.display_name LIKE '%" . esc_sql( $s ) . "%'
			OR um.meta_value LIKE '%" . esc_sql( $s ) . "%'
		)
	";

	/**
	 * Allow filtering the member directory search SQL to be used.
	 *
	 * @since TBD
	 *
	 * @param string $sql_where_search The member directory search SQL to be used.
	 * @param string $search_text      The search text used.
	 */
	$sql_where_search = apply_filters( 'pmpro_member_directory_search_sql', $sql_where_search, $s );

	$sql_parts['WHERE'] .= $sql_where_search;
```